NEWAPI.md

```diff
@@ -0,0 +1,202 @@
+
+- Your `generateapis.sh` change
+- The new directory under `apis`
+
+Push this change up to your fork on github. Don't create a PR yet.
+
+Step 7: Pull your change on Windows
+-----------------------------------
+
+Clone your fork on Windows if you haven't already, and pull your
+change. (I find `git fetch --all` handy here, then just checkout
+your new branch.)
+
+Step 8: Modify the API catalog
+------------------------------
+
+Edit `apis/apis.json`. Again, this is in alphabetical order - please keep it that way.
+You'll typically want JSON like this:
+
+```json
+{
+  "id": "FIXME",
+  "productName": "FIXME",
+  "productUrl": "FIXME",
+  "version": "1.0.0-alpha00",
+  "type": "grpc",
+  "description": "FIXME",
+  "tags": [ "FIXME_1", "FIXME_2" ],
+}
+```
+
+Fix everything with "FIXME". There's no set number of tags, but these are used for the NuGet package,
+so consider what users will search for.
+
+The version number of "1.0.0-alpha00" is used to indicate that there
+hasn't been a release of this API yet. It will be incremented to
+`1.0.0-alpha01` as part of the first release.
+
+If your project uses the IAM or long-running operations APIs, you'll need to add dependencies for those, e.g.
+
+```json
+"dependencies": {
+  "Google.LongRunning": "project",
+  "Google.Cloud.Iam.V1": "project"
+}
+```
+
+Look at other APIs for example values.
+
+Step 9: Generate project and solution files
+-------------------------------------------
+
+Run `generateprojects.sh`. This should create:
+
+- A solution file
+- Project files for the production project and the snippets
+
+If you have Visual Studio 2017 installed, now would be a good time
+to load the solution and check it at least builds.
+
+(Note: currently if your API uses long-running operations, it *won't* compile until you've added a partial class like [this](https://github.com/GoogleCloudPlatform/google-cloud-dotnet/blob/58a38a605df48f8ee70d36e41c948bbd1b3628f3/apis/Google.Cloud.Speech.V1/Google.Cloud.Speech.V1/CloudSpeechGrpcPartial.cs).
+We will be generating that code soon.)
+
+Step 10: Commit changes
+-----------------------
+
+Commit the change to `apis.json` and the generated files.
+
+Optional: Step 11: Create documentation
+---------------------------------------
+
+We typically make sure there's documentation available before we
+build any NuGet packages. If you intend to release a first version
+soon, it's worth creating a simple documentation page right from the
+start.
+
+- Create a `docs` directory under your API directory (e.g. `apis\Google.Cloud.Datastore.V1\docs`)
+- Create an `index.md` file under the `docs` directory. Start off with content like this (verbatim):
+
+```
+{{title}}
+
+{{description}}
+
+{{installation}}
+
+{{auth}}
+```
+
+- Ideally, add a "getting started" section indicating the client
+  classes that act as an entry point to the API.
+- Even more ideally, add some examples in the Snippets project and embed them in the docs.
```

Comment:
Happy to make it another step, sure. Maybe I'll reorder things so that it's "Write a snippet to test things if you possibly can" followed by "Write docs that show your snippet if you wrote one".